src/Microsoft.AspNetCore.Rewrite/Internal/ModRewrite/FlagParser.cs

```diff
@@ -2,22 +2,68 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Generic;
 
 namespace Microsoft.AspNetCore.Rewrite.Internal.ModRewrite
 {
     public class FlagParser
-    { 
-        public static Flags Parse(string flagString)
+    {
+        private IDictionary<string, FlagType> _ruleFlagLookup = new Dictionary<string, FlagType>(StringComparer.OrdinalIgnoreCase) {
+            { "b", FlagType.EscapeBackreference},
+            { "c", FlagType.Chain },
+            { "chain", FlagType.Chain},
+            { "co", FlagType.Cookie },
+            { "cookie", FlagType.Cookie },
+            { "dpi", FlagType.DiscardPath },
+            { "discardpath", FlagType.DiscardPath },
+            { "e", FlagType.Env},
+            { "env", FlagType.Env},
+            { "end", FlagType.End },
+            { "f", FlagType.Forbidden },
+            { "forbidden", FlagType.Forbidden },
+            { "g", FlagType.Gone },
+            { "gone", FlagType.Gone },
+            { "h", FlagType.Handler },
+            { "handler", FlagType.Handler },
+            { "l", FlagType.Last },
+            { "last", FlagType.Last },
+            { "n", FlagType.Next },
+            { "next", FlagType.Next },
+            { "nc", FlagType.NoCase },
+            { "nocase", FlagType.NoCase },
+            { "ne", FlagType.NoEscape },
+            { "noescape", FlagType.NoEscape },
+            { "ns", FlagType.NoSubReq },
+            { "nosubreq", FlagType.NoSubReq },
+            { "p", FlagType.Proxy },
+            { "proxy", FlagType.Proxy },
+            { "pt", FlagType.PassThrough },
+            { "passthrough", FlagType.PassThrough },
+            { "qsa", FlagType.QSAppend },
+            { "qsappend", FlagType.QSAppend },
+            { "qsd", FlagType.QSDiscard },
+            { "qsdiscard", FlagType.QSDiscard },
+            { "qsl", FlagType.QSLast },
+            { "qslast", FlagType.QSLast },
+            { "r", FlagType.Redirect },
+            { "redirect", FlagType.Redirect },
+            { "s", FlagType.Skip },
+            { "skip", FlagType.Skip },
+            { "t", FlagType.Type },
+            { "type", FlagType.Type },
+        };
+
+        public Flags Parse(string flagString)
         {
             if (string.IsNullOrEmpty(flagString))
             {
-                return null;
+                throw new ArgumentNullException("flagString is null");
             }
 
             // Check that flags are contained within []
             if (!flagString.StartsWith("[") || !flagString.EndsWith("]"))
             {
-                throw new FormatException();
+                throw new FormatException("Flags do no start or end with square brackets: [flags]");
```

Comment:
Not a fan of this grammar
"Flags should start and end with square brackets: [flags]"?